### ISSUE TITLE
[Bug] Wrong jakarta OSGi imports in main branch

### ISSUE THREAD

**Issue by marcanpilami** [apache/shiro#1449]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/shiro/issues?q=is%3Aissue) and found no similar issues.

### Environment

Apache Felix, OpenJDK 11 or 21

### Shiro version

2.0.1-SNAPHOT (built by myself) 

### What was the actual outcome?

Cannot start the shiro-web bundle. This is a follow-up to #1324 (with a few comments, including a discussion on possible fixes).

The 1324 fix seems to be incomplete: it replaces the javax namespace by a jakarta one inside the manifest, but without changing the version of the import. 

For example, the normal package imports `javax.annotation.security;version="[1.3,2)"`. The fix for the jakarta version just renames it to `jakarta.annotation.security;version="[1.3,2)"`.

However, Jakarta never released a 1.3.x bundle containing jakarta.* package - they only released a transition bundle containing javax.* packages. The first version of jakarta.annotations-api containing a jakarta.* package is 2.0.0. 

So we need to change the version, not only the imports. I do not know what shade is able to do here - we may have to add a Maven profile to change the dependencies. 

### What was the expected outcome?

The bundle starts.

### How to reproduce

Just try to start it in any OSGi container like Karaf.

**Comment by lprimak** [comment]
Is there a proper way to just use `version 1.3 or above` in OSGi? Or `1.3 or 2 or above`
So we wouldn't have to "fix" a particular version unnecessarily?

**Comment by fpapon** [comment]
I'm playing with the shade maven plugin to make it work.

**Comment by lprimak** [comment]
This may be a source of your issues @steinarb

**Comment by steinarb** [comment]
@lprimak  This is probably an issue for me too, but not related to the classloader issue mentioned earlier (probably the next thing I will run into once the classloader issue has a fix... :smile: )

@fpapon Noted! I will see what I can find out when I get to it.

**Comment by lprimak** [comment]
I think 3.0 would be released before this was ever completed

**Comment by steinarb** [comment]
@lprimak I think maybe it is closing in on something I have been procrastinating on for years: finding out the state of, and plans for, javax.servlet vs jakarta.servlet for pax-web and karaf with versions and plans (for each major upgrade for the last couple of year I've been like "Oh, still works! Ok, later...").

If I write up a summary, is there a suitable place I should put it? 
In a new issue, perhaps, with a very specific title related to OSGi and javax.servlet to jakarta.servlet?

**Comment by fpapon** [comment]
Agreed with @lprimak the jakarta import are fixed on 3.x branch and will be release soon.

**Comment by lprimak** [comment]
@steinarb Somewhere on the Shiro web site would be great. Blog on Shiro site, Blog of foojay.io would work.
